### PR TITLE
HTTP/3 support for k6

### PR DESCRIPTION
**Pull request by bandorko**
## What?

HTTP/3 support for k6

## Why?

This is a proposal for implementing HTTP/3 support for K6. I know that it is incomplete (e.g.: there are no test), but I'm really interested in your thoughts on this approach. It uses a ConnectionTracker to collect the metrics about the communication.
I also created it as an extension (https://github.com/bandorko/xk6-http3), but because its code is mostly copied from the k6 source, I thought that HTTP/3 implementation would have a better place in the k6 repository.

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)
https://github.com/grafana/k6/issues/238

### CODE REVIEW

File: lib/netext/httpext/http3.go
Comment (codebien) [inline lib/netext/httpext/http3.go:43]: I don't think we need to define new metrics. We could reuse the already defined for the current HTTP module, of course, the proto tag should be different.

**Review by codebien — CHANGES_REQUESTED** [review]
Hey @bandorko,
thanks for your contribution :tada: I appreciate the fact you want to discuss with us the vision of the new API.

As we always suggest and similar to the process we use for building built-in modules, the process for including a new API is to have an extension where we start gathering feedback and usage--you already did it :handshake: 
When it gets enough traction to justify being included in the core then we start the process.

However, regarding the HTTP proposal. We already have internally the plan to create a new HTTP module, and we added a new design doc explaining our vision around it. So, it sounds like a better approach if an eventual new protocol is implemented following the design we described there.

Regarding the HTTP3 version, the Go team is working on supporting native in Go the 3rd version of the HTTP protocol. Considering it, I think it makes sense to wait for this release before including it directly into the core. In the meantime, the Core team and the community can continue to experiment with other libraries in the extension.

**Comment by oleiade** [comment]
I fully align with @codebien points 🙇🏻

**Comment by bandorko** [comment]
@codebien, @oleiade: Thank you for the feedback. Then I close this PR and fix the metric names in the extension until http3 will be supported in go and k6.
Do you have a link for the plan of the new HTTP module, or it is still internal?

**Comment by codebien** [comment]
@bandorko you can find the design doc [here](https://github.com/grafana/k6/blob/master/docs/design/018-new-http-api.md). If you would like to align and you need support, feel free to open an issue directly on the extension and ping me.

**Pull request closed without merging**
